### PR TITLE
Update test.yml

### PR DESCRIPTION
**Pull request by willfrey**
- Adds 3.8 to the `SUPPORTED_PYTHON_VERSIONS`
- Adds 1.1.10 to `SUPPORTED_POETRY_VERSIONS`
- Adds snok/install-poetry@v1.2 to `test-major-and-minor-versions`
- Updates `LATEST_POETRY` to 1.1.10
- Updates hard-coded poetry version in `test-latest-version-when-unspecified` to 1.1.10
- Reformat `${{fromJson(...)}}` -> `${{ fromJson(...) }}` for consistency
- Removes trailing `-` from `test-install` `name` attribute

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,13 +8,13 @@ on: pull_request
 
 env:
   # JSON variables (used in our strategy/matrix)
-  SUPPORTED_POETRY_VERSIONS: '\"poetry-version\":[\"1.1.0\", \"1.1.8\", \"1.2.0a2\"]'
-  SUPPORTED_PYTHON_VERSIONS: '\"python-version\":[\"3.7\", \"3.9\", \"3.10.0-beta.4\"]'
+  SUPPORTED_POETRY_VERSIONS: '\"poetry-version\":[\"1.1.0\", \"1.1.8\", \"1.1.10\", \"1.2.0a2\"]'
```
Comment (sondrelg) [inline .github/workflows/test.yml:11]:
```suggestion
  SUPPORTED_POETRY_VERSIONS: '\"poetry-version\":[\"1.1.0\", \"1.1.10\", \"1.2.0a2\"]'
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -8,13 +8,13 @@ on: pull_request
...
+  SUPPORTED_PYTHON_VERSIONS: '\"python-version\":[\"3.7\", \"3.8\", \"3.9\", \"3.10.0-beta.4\"]'
```
Comment (sondrelg) [inline .github/workflows/test.yml:12]:
```suggestion
  SUPPORTED_PYTHON_VERSIONS: '\"python-version\":[\"3.7\", \"3.9\", \"3.10.0-rc.2\"]'
```

**Review by sondrelg — COMMENTED** [review]
This is great - thanks @willfrey!

Before merging this, would you mind removing the poetry and python versions? We recently pruned the list because tests were taking waaayy too long. Anything that fails on 3.8 because of new syntax should fail and 3.7, and `1.1.8` is probably not needed if we're testing `1.1.10` 🙂 

Everything else looks good 👍

**Comment by willfrey** [comment]
Done!

**Comment by sondrelg** [comment]
Great, thanks!

**Pull request merged**